api/overriders.js

```diff
@@ -0,0 +1,165 @@
+var _ = require('underscore'),
+    fs = require('fs'),
+    mkdirp = require('mkdirp'),
+    path = require('path'),
+    folders = require('./folders'),
+    bodyParser = require('body-parser');
+
+function joinIfArray(folder) {
+    if (Array.isArray(folder)) {
+        return folder.join(',');
+    }
+    return folder;
+}
+
+function Overrides() {
+    this.save = {
+        folders: function(proxy, route) {
+            if (proxy.override && route.overrides) {
+                if (Array.isArray(proxy.override) && proxy.override.length > 0) {
+                    _.each(proxy.override, function(key, index) {
+                        var override = route.overrides[key];
+                        var name = override.name || override.folder || index || 'override';
+                        override.name = path.join(proxy.name, name);
+                        folders[override.name] = folders[proxy.name];
+                        if (override && override.folder) {
+                            folders[override.name] = path.join(folders[route.name], override.folder);
+                            if (!fs.existsSync(folders[override.name])) {
```

Comment:
Again, a duplicate logic - saw this before several times. Let's offload it to a utility class / method.